scuttlebutt/src/lib.rs

```diff
@@ -39,15 +39,53 @@ use tracing::{debug, error};
 use crate::digest::Digest;
 use crate::message::ScuttleButtMessage;
 use crate::serialize::Serializable;
-pub use crate::state::ClusterState;
+pub use crate::state::{ClusterState, SerializableClusterState};
 use crate::state::NodeState;
 
 /// Map key for the heartbeat node value.
 pub(crate) const HEARTBEAT_KEY: &str = "heartbeat";
 
 pub type Version = u64;
 
-pub type NodeId = String;
+// pub type NodeId = String;
+
+#[derive(Default, Clone, Hash, Eq, PartialEq, PartialOrd, Ord, Debug, Serialize, Deserialize)]
+pub struct NodeId {
+    id: String,
+    gossip_public_address: String,
```

Comment:
why did you remove `node_generation`

Reply:
It will be available from the client-side (Quickwit). See the docs
while writing the docs, I noticed we can just use `{node_unique_id}/{node_generation}` instead of `{node_unique_id}/{gossip_public_address}/{generation}`
https://github.com/quickwit-oss/scuttlebutt/pull/22/files#diff-9a1c8c0b2fe22d61843aec8d062321324f330717737a4880c634fc3c5a7b43fbR74